crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingControllerTest.java

```diff
@@ -40,10 +33,6 @@
 class CrownCourtProceedingControllerTest {
```

Comment:
I think it would be better to avoid spinning up the whole application context here. I tested with the following code and not only did the tests run much faster, it also meant a lot of boilerplate could be removed.


```java

@DirtiesContext
@AutoConfigureMockMvc(addFilters = false)
@WebMvcTest(CrownCourtProceedingController.class)
class CrownCourtProceedingControllerTest {

    private static final boolean IS_VALID = true;
    private static final String ENDPOINT_URL = "/api/internal/v1/proceedings";

    @Autowired
    private MockMvc mvc;

    @Autowired
    private ObjectMapper objectMapper;

    @MockBean
    private ProceedingService proceedingService;
```
This would effectively remove the spring security filters, which I think is ok in this case because the main focus is ensuring the Controller works as intended. It would also allow us to remove the tests that check for a 401/403 responses, but I think as long as this is being covered in the integration tests, we're covering all of our bases. In fact, i feel like there's quite a lot of duplication in the tests at the moment.

Reply:
Agree. We can disable security for controller test. I noted that the same test scenarios has been covered for HTTP Methods - GET, PUT, POST. I feel it is OK to be repeated.